src/evict/evict_page.c

```diff
@@ -179,9 +179,17 @@ __evict_delete_ref(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 		 * something is busy, be sure that the page still ends up
 		 * marked deleted.
 		 */
-		if (ndeleted > pindex->entries / 10 && pindex->entries > 1 &&
-		    (ret = __wt_split_reverse(session, ref)) != EBUSY)
-			return (ret);
+		if (ndeleted > pindex->entries / 10 && pindex->entries > 1) {
+			if ((ret = __wt_split_reverse(session, ref)) == 0)
+				return (0);
```

Comment:
@michaelcahill, thanks! I think you're correct.

Pulling up the error message means fixing `__split_multi` as well as `__split_reverse`, and I'm hesitant to do that, I'd prefer `__split_parent` own the problem of deciding if the split was successful or not. We could push setting `WT_REF_DELETED` back down into `__split_parent` again, but that's ugly, too.

I pushed a change to have that case return `EBUSY` (and, as you noted, suppressing the error message because it's an expected condition), that change is 846125e.

Two other things occurred to me:

* Could we set the WT_REF we're deleting to `WT_REF_DELETED` before calling `__split_parent`? If so, then we don't have to handle the `EBUSY` error return at all. I've pushed a branch (#2309) to show you what I mean.
* Could we continue the internal page reverse split, even if the result would be an empty page, by leaving one slot in the WT_REF_DELETED state?  I've pushed a branch (#2310) to show you what I mean.